File: controller/merchant_controller.py
from util import Util
from usecases.merchant_usecase import MerchantUseCase
from datastore.local_datastore import LocalDataStore


class MerchantController:
    def __init__(self):
        self._merchant_use_case = MerchantUseCase(LocalDataStore())

    @Util.check_no_of_args(Util.get_function_params_list(MerchantUseCase.new_merchant))
    def new_merchant(self, *args):
        return self._merchant_use_case.new_merchant(*args)

    @Util.check_no_of_args(Util.get_function_params_list(MerchantUseCase.update_merchant_discount_percentage))
    def update_merchant_discount_percentage(self, *args):
        return self._merchant_use_case.update_merchant_discount_percentage(*args)

    @Util.check_no_of_args(Util.get_function_params_list(MerchantUseCase.report_discount))
    def report_discount(self, *args):
        return self._merchant_use_case.report_discount(*args)

File: java/src/main/java/com/paylater/controller/MerchantController.java
package com.paylater.controller;

import com.paylater.repository.LocalDatabase;
import com.paylater.usecases.MerchantUseCase;
import com.paylater.utils.ResponseValue;

public class MerchantController {
    private final MerchantUseCase merchantUseCase;

    public MerchantController() {
        this.merchantUseCase = new MerchantUseCase(LocalDatabase.getInstance());
    }

    public ResponseValue newMerchant(String[] args) {
        String name = args[0];
        String email = args[1];
        String discountPercentage = args[2];

        return merchantUseCase.newMerchant(name, email, discountPercentage);
    }

    public ResponseValue updateMerchantDiscountPercentage(String[] args) {
        String merchant = args[0];
        String percentage = args[1];

        return merchantUseCase.updateMerchantDiscountPercentage(merchant, percentage);
    }

    public ResponseValue reportDiscount(String[] args) {
        String merchant = args[0];
        return merchantUseCase.reportDiscount(merchant);
    }
}
